### PR TITLE
add missing params to units.yml

### LINKED ISSUE #184: units.yml has been updated to ensure all existing params are included

### PR DESCRIPTION
**Pull request by jmcook1186**
### Types of changes

- [x] Enhancement (project structure, spelling, grammar, formatting)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### A description of the changes proposed in the Pull Request

Adds missing parameters to `units.yml` and organizes into alphabetical order
Updates term `processor` to `physical-processor` everywhere.

Fixes #184

### CODE REVIEW

File: src/config/units.yaml [hunk 1]
```diff
@@ -18,43 +18,116 @@ duration:
...
+  aggregation: sum
+operational-carbon:
+  description: Operational Emissions of the component
+  unit: gCO2e
+  aggregation: sum
+pe.use:
+  description: Primary energy associated with component use (returned from Boavizta)
+  unit: kWh
+  aggregation: sum
+processor:
```
Comment (gnanakeethan) [inline src/config/units.yaml:93]: `phyiscal-processor` as per latest changes in tdp-finder. 

I think we need to change to that in the other models.

File: src/config/units.yaml [hunk 2]
```diff
@@ -18,43 +18,116 @@ duration:
...
+  aggregation: avg
+ram-util:
+  description: refers to percentage of memory utilized.
+  unit: percentage
+  aggregation: avg
+rr:
+  description: resources reserved for an application
+  unit: count
+  aggregation: none
+tdp:
```
Comment (jawache) [inline src/config/units.yaml:113]:
```suggestion
thermal-design-power:
```

In keeping with the philosophy of expanding acronyms we should probably expand this too

File: src/config/units.yaml [hunk 3]
```diff
@@ -18,43 +18,116 @@ duration:
...
+  description: thermal design power for a processor
+  unit: kwh
+  aggregation: avg
+te:
```
Comment (jawache) [inline src/config/units.yaml:117]:
```suggestion
total-embodied-emissions:
```

File: src/config/units.yaml [hunk 4]
```diff
@@ -18,43 +18,116 @@ duration:
...
+  description: total embodied emissions of some component
+  unit: gCO2e
+  aggregation: sum
+timestamp:
+  description: refers to the time of occurrence of the observation
+  unit: RFC3339
+  aggregation: None
+tir:
```
Comment (jawache) [inline src/config/units.yaml:125]:
```suggestion
time-reserved:
```

File: src/config/units.yaml [hunk 5]
```diff
@@ -18,43 +18,116 @@ duration:
...
-  aggregation: sum
-energy:
-  description: amount of energy utilised by the component
-  unit: kWh
-  aggregation: sum
 e-cpu:
   description: Energy consumed by the CPU of the component
   unit: kWh
   aggregation: sum
+el:
```
Comment (jawache) [inline src/config/units.yaml:25]:
```suggestion
expected-lifespan:
```

File: src/config/units.yaml [hunk 6]
```diff
@@ -18,43 +18,116 @@ duration:
...
+  description: expected lifespan for some component
+  unit: seconds
+  aggregation: none
 e-mem:
```
Comment (jawache) [inline src/config/units.yaml:29]:
```suggestion
energy-memory:
```

Spirit of using long descriptive names

File: src/config/units.yaml [hunk 7]
```diff
@@ -18,43 +18,116 @@ duration:
...
-  aggregation: None
-carbon:
-  description: an amount of carbon emitted into the atmosphere
-  unit: gCO2e
-  aggregation: sum
-energy:
-  description: amount of energy utilised by the component
-  unit: kWh
-  aggregation: sum
 e-cpu:
```
Comment (jawache) [inline src/config/units.yaml:21]:
```suggestion
energy-cpu:
```

File: src/config/units.yaml [hunk 8]
```diff
@@ -18,43 +18,116 @@ duration:
...
   unit: gCO2e
   aggregation: sum
-operational-carbon:
-  description: Operational Emissions of the component
-  unit: gCO2e
+energy:
+  description: amount of energy utilised by the component
+  unit: kWh
+  aggregation: sum
+e-net:
```
Comment (jawache) [inline src/config/units.yaml:41]:
```suggestion
energy-net:
```

File: src/config/units.yaml [hunk 9]
```diff
@@ -18,43 +18,116 @@ duration:
...
+  aggregation: none
+functional_unit_time:
+  description: string describing the unit of time in which the final SCI calculation should be expressed
+  unit: none
+  aggregation: none
+gpu-util:
+  description: refers to CPU utilization.
+  unit: percentage
+  aggregation: avg
+grid-ci:
```
Comment (jawache) [inline src/config/units.yaml:65]:
```suggestion
grid-carbon-intensity:
```

I can't in my head justify keeping grid-ci (which I admit I named) and at the same time expanding some of our other shorter acronyms out.

**Review by jawache — CHANGES_REQUESTED** [review]
So after reviewing I think we are evolving a standard for naming.

- Use hyphens not underscores.
- Lowercase
- Acronyms are not allowed, must use long form for naming.
- Shortening long names to just the first character is not allowed, e.g. e-cpu, must be energy-cpu.
- Common terms that are just shortened forms of long names **are** allowed (e.g. util instead of utilization) - this last one is dodgy, but let's see how it goes!

File: src/config/units.yaml [hunk 10]
```diff
@@ -18,43 +18,116 @@ duration:
...
+  aggregation: None
+manufacture:
+  description: Carbon emitted from component manufacture (returned from Boavizta)
+  unit: gCO2eq
+  aggregation: sum
+operational-carbon:
+  description: Operational Emissions of the component
+  unit: gCO2e
+  aggregation: sum
+pe.use:
```
[the remark this answers is not in the document]
Comment (jmcook1186) [reply src/config/units.yaml:89]: Reticent to change this as the only places we use it is as a key to parse data from the boavizta API response, where it is labelled `pe` - we only use it as an intermediate value that enables us to report `energy`. 

So it's not really renamable. Maybe it shouldn't even be included.

File: src/config/units.yaml (hunk 8 above)
Comment (jmcook1186) [reply src/config/units.yaml:41]: using `energy-network` for consistency with `energy-memory`

**Comment by jawache** [comment]
Looks good to me!

**Pull request merged**